scheduler/plugin/guest.go

```diff
@@ -220,6 +220,11 @@ func (g *guest) normalizeScore(ctx context.Context) (framework.NodeScoreList, *f
 	statusCode := int32(callStack[0])
 	statusReason := paramsFromContext(ctx).resultStatusReason
 	normalizedScoreList := paramsFromContext(ctx).resultNormalizedScoreList
+	if len(normalizedScoreList) == 0 {
```

Comment:
nit: What if the length of `normalizedScoreList` is not equal to the length of `nodeScoreList` ?

Reply:
We return the error:
https://github.com/kubernetes-sigs/kube-scheduler-wasm-extension/blob/main/scheduler/plugin/plugin.go#L341-L343

We shouldn't restore the score list in that case because if the length is unmatched, that's the bug in the guest. (The guest filled in an invalid node score list) 
We should error out, rather than hiding the bug.

And, what we're trying to do here is not hide the bug. When the guest doesn't implement the normalizescore, we just return without doing anything, which results in an empty `resultNormalizedScoreList`, and hence fail.

https://github.com/kubernetes-sigs/kube-scheduler-wasm-extension/blob/main/guest/scoreextensions/scoreextensions.go#L64-L68

Reply:
> what we're trying to do here is not hide the bug. When the guest doesn't implement the normalizescore, we just return without doing anything, which results in an empty resultNormalizedScoreList, and hence fail.

Make sense. Thank you so much for your explanation!